### PR TITLE
Fix module folder for future checkmk releases

### PR DESCRIPTION
**Pull request by msekania**
Fixes issue #389 

## Pull request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Currently all three options *attributes*, *remove_attributes*, and *update_attributes* can be used simultaneously, which is no longer supported by API for Check MK >= v2.2.0p7

Issue Number: #389 

## What is the new behavior?

- issue a warning message if more then one option from *attributes*, *remove_attributes*, and *update_attributes* is used simultaneously for Check MK  < v2.2.0p7 
- fail with error message if more then one option from *attributes*, *remove_attributes*, and *update_attributes* is used simultaneously for Check MK  >= v2.2.0p7

### CODE REVIEW

**Comment by robin-checkmk** [comment]
@msekania thanks for tackling this!
Is your branch up-to-date with our `devel`? Looks like there are some unrelated changes in here.

**Comment by msekania** [comment]
> Is your branch up-to-date with our `devel`? Looks like there are some unrelated changes in here.

apparently not, changes should be only in folder.py

**Comment by robin-checkmk** [comment]
I'll take a look at the failing tests and the weirdness with the additional changes.

**Comment by msekania** [comment]
@robin-checkmk 

thanks!

**Comment by robin-checkmk** [comment]
I understand the issue with the integration tests now but the solution is not trivial.
I will update this comment, once I have news.

**Comment by robin-checkmk** [comment]
@msekania I implemented a hotfix for the integration tests, but the actual fix is already in `devel`.
So the current test findings should be valid, and if you pull the `devel` branch into yours, the integration tests should be fixed properly.

File: plugins/modules/folder.py [hunk 1]
```diff
@@ -333,6 +361,62 @@ def run_module():
...
+            )
+            or (
+                int(checkmkversion[0]) == 2
+                and int(checkmkversion[1]) == 2
+                and int(patchlevel[0]) == 0
+                and patchtype == "p"
+                and int(patchlevel[1]) >= 7
+            )
+        ):
+            version_ge_220p7 = True
```
Comment (lgetwan) [inline plugins/modules/folder.py:408]: Can you put this into a separate function to keep the run_module() function a bit shorter?
BTW: This algorithm will not work for daily build versions, but I won't consider this a show stopper. ;-)

**Review by lgetwan — CHANGES_REQUESTED** [review]
Looks fine! Thanks for your contribution.
I just added one comment that you should solve before we merge this.

**Comment by msekania** [comment]
@robin-checkmk,

I pulled devel branch, and merged it in the current, which required rebase.

**Comment by robin-checkmk** [comment]
@msekania I am uncertain what exactly happened, but I think your idea of creating a new and clean PR ist the best approach. Go ahead and create it, I will merge tomorrow then and release.
Thanks!

**Pull request closed without merging**